src/App.jsx

```diff
@@ -1,24 +1,221 @@
-  // Destructure variables, properties and methods from the useQRCodeGenerator hook that you imported above here :)
+// *** Styled components *** //
+const AppMain = styled.div`
+background-color: #FFD2CD;
+display: flex;
+flex-direction: column;
+justify-content: center;
+align-items: center;
+height: 100vh;
+text-align: center;
+gap: 2%;
+
+@media (min-width: 800px) {
+  gap: 1%;
+}
+`
+
+const QRPlant = styled.img`
+width: 50%;
+
+@media (min-width: 800px) {
+  width: 40%;
+}
+
+@media (min-width: 1000px) {
+  width: 20%;
+}
+`
+
+const Titles = styled.div`
+  position: center;
+  display: inline-block;
+`
+
+const H1 = styled.h1`
+  color: #4B634D;
+  font-size: 2rem;
+  text-align: center;
+  margin: 5% 0% 3% 0%;
+`
+
+const H2 = styled.h2`
+  color: #4B634D;
+  font-size: 1,5rem;
+`
+
+const Button = styled.button`
+  background-color: #5F7761;
+  color: white;
+  padding: 10px 25px;
+  text-align: center;
+  font-size: 1rem;
+  margin: 2px;
+  cursor: pointer;
+  border-radius: 20px;
+  border: none;
+`
+
+const FormContainer = styled.form`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  margin-bottom: 20px;
+`
+
+const InputField = styled.input`
+  padding: 8px;
+  margin-bottom: 5%;
+  border: 1px solid #4B634D;
+  font-size: 1rem;
+`
+
+// Styling of error message if the URL isn't correct
+const ErrorMessage = styled.div`
+  color: #4B634D;
+  margin-top: 8px;
+  font-size: 14px;
+`
+
+// Styling of the generated code
+const QRCodeImage = styled.img`
+  display: block;
+  margin: auto; /* This will center the image horizontally */
+  margin-top: 3%;
+  margin-bottom: 5%;
+
+  @media (min-width: 1000px) {
+    margin-top: 1%;
+    margin-bottom: 1%;
+  }
+`
+
+// Styling of the dialogbox that opens after QR-code is generated if user wants to download
+const DialogBox = styled.div`
+  position: fixed;
+  z-index: 1000;
+  left: 0;
+  top: 0;
+  width: 100%;
+  height: 100%;
+  background-color: rgba(0, 0, 0, 0.5);
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`
+
+const DialogContent = styled.div`
+  background: white;
+  padding: 10%;
+`
+
+// App component definition
+const App = () => {
+  // Using the custom hook to manage QR code generation logic
+  const { url, setUrl, qrCode, showInput, generateQRCode, resetQRCode } = useQRCodeGenerator()
+  // State variables managed by the component
+  const [showDialogbox, setShowDialogbox] = useState(false)
+  const [fileName, setFileName] = useState('')
+  const [errorMessage, setErrorMessage] = useState('')
+
+  // Function to validate the URL and generate the QR code
+  const handleGenerate = (e) => {
+    e.preventDefault();
+    const validUrlRegex = /^(https?:\/\/)?([\da-z.-]+)\.([a-z.]{2,6})([/\w .-]*)*\/?$/
+
+    if (validUrlRegex.test(url)) {
+      generateQRCode(url)
+      setErrorMessage('')
+    } else {
+      setErrorMessage("Oups! Enter a valid web address (hint: https://www...)")
```

Comment:
Cool validation! 😊
Did you know though, that QR-codes can include anything, not only links to web-sites? Blocking everything that's not an URL limits the usage of it. :)